packages/editor/src/browser/editor-command.ts

```diff
@@ -35,10 +38,54 @@ export namespace EditorCommands {
 @injectable()
 export class EditorCommandContribution implements CommandContribution {
 
+    constructor(@inject(FileSystem) protected readonly fileSystem: FileSystem,
+                @inject(PreferenceService) protected readonly preferences: PreferenceService) {}
+
     registerCommands(registry: CommandRegistry): void {
         registry.registerCommand(EditorCommands.SHOW_REFERENCES);
         registry.registerCommand(EditorCommands.CONFIG_INDENTATION);
         registry.registerCommand(EditorCommands.INDENT_USING_SPACES);
         registry.registerCommand(EditorCommands.INDENT_USING_TABS);
+
+        registry.registerCommand(CommonCommands.ENABLE_AUTO_SAVE, {
+            isEnabled: () => !this.isAutoSaveOn(),
+            execute: () => this.switchAutoSave()
+        });
+        registry.registerCommand(CommonCommands.DISABLE_AUTO_SAVE, {
+            isEnabled: () => this.isAutoSaveOn(),
+            execute: () => this.switchAutoSave()
+        });
+    }
+
+    private async switchAutoSave(): Promise<void> {
+        const home = await this.fileSystem.getCurrentUserHome();
```

Comment:
regardless of the requirement question, this is not the single source of preferences. have a look at: https://github.com/theia-ide/theia/blob/master/packages/preferences/src/browser/preference-frontend-module.ts#L20